Modules/AnytypeCore/AnytypeCore/Utils/TaskGroup.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+fileprivate struct SortedItem<ChildTaskResult> {
+    let index: Int
+    let content: ChildTaskResult
+}
+
+public struct SortedTaskGroup<ChildTaskResult> where ChildTaskResult : Sendable {
+    
+    private var index: Int = 0
+    
+    fileprivate var taskGroup: TaskGroup<SortedItem<ChildTaskResult>>
+    
+    fileprivate init(taskGroup: TaskGroup<SortedItem<ChildTaskResult>>) {
+        self.taskGroup = taskGroup
+    }
+    
+    public mutating func addTask(priority: TaskPriority? = nil, operation: @escaping @Sendable () async -> ChildTaskResult) {
+        let currentIndex = index
+        taskGroup.addTask(priority: priority) { SortedItem(index: currentIndex, content: await operation()) }
+        index+=1
```

Comment:
nit
looks like we don't need `let currentIndex` here, just use `index`
also I really want to add some spaces here `index += 1` ))

Reply:
needed for capture current value in async task `SortedItem(index: currentIndex`

Reply:
Spaces fixed